v2/test/integration/http/loopback_v03_test.go

```diff
@@ -153,6 +155,39 @@ func TestClientLoopback_binary_base64_v03tov03(t *testing.T) {
 				ContentLength: 32,
 			},
 		},
+		"Loopback v0.3 with error": {
+			now: now,
+			event: &cloudevents.Event{
+				Context: cloudevents.EventContextV1{
+					ID:              "ABC-123",
+					Type:            "unit.test.client.sent",
+					Source:          *cloudevents.ParseURIRef("/unit/test/client"),
+					Subject:         strptr("resource"),
+					DataContentType: cloudevents.StringOfApplicationJSON(),
+				}.AsV1(),
+				DataEncoded: toBytes(map[string]interface{}{"hello": "unittest"}),
+			},
+			result: cloudevents.NewHTTPResult(http.StatusBadRequest, "unit test %s", http.StatusText(http.StatusBadRequest)),
```

Comment:
Because 400 could be fired directly by the sdk internals, can you use whatever other status code to check that the status code you wrote as result is exactly the one you receive on the other side?

Reply:
We can follow up with more tests for more status codes. Let's let this test in and add more.

Reply:
I'm not saying to add more tests for status codes, i'm just saying to change this because this status code could be triggered by the sdk and not by the user, while the test here is testing that the status code triggered by the user is written on the wire

Reply:
Found out why result assertion doesn't work. We didn't set return status code in the `TapHandler`. Fixed that and a bunch of related tests. Trying to keep a minimum change in this PR. So please don't suggest any behavioral change in this PR.